script/hassfest/coverage.py

```diff
@@ -141,23 +142,29 @@ def validate(integrations: dict[str, Integration], config: Config) -> None:
 def generate(integrations: dict[str, Integration], config: Config) -> None:
     """Sort coverage."""
     coverage_path = config.root / ".coveragerc"
-    lines = []
-    start = False
+    core = []
+    components = []
+    section = "header"
 
     with coverage_path.open("rt") as fp:
         for line in fp:
-            if (
-                not start
-                and line
-                == "    # omit pieces of code that rely on external devices being present\n"
-            ):
-                start = True
-            elif line == "[report]\n":
+            if line == "[report]\n":
                 break
-            elif start and line != "\n":
-                lines.append(line)
 
-    content = f"{PREFIX}{"".join(sorted(lines))}\n\n{SUFFIX}"
+            if section != "core" and line == "omit =\n":
+                section = "core"
+            elif section != "components" and line == COMPONENTS_PREFIX:
+                section = "components"
+            elif section == "core" and line != "\n":
+                core.append(line)
+            elif section == "components" and line != "\n":
+                components.append(line)
+
```

Comment:
The above logic is fairly susceptible to minor trip-ups such as spaces in the wrong place. Given that we're about to overwrite the user's file including any modifications, I suggest a sanity check here.
```suggestion
assert(core, "core should be a non-empty list")
assert(components, "components should be a non-empty list")
```